mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -518,7 +518,7 @@ struct LockedAmountInfo {
 }
 
 struct SwapsContext {
-    running_swaps: Mutex<Vec<Weak<dyn AtomicSwap>>>,
+    running_swaps: Mutex<HashMap<Uuid, Arc<dyn AtomicSwap>>>,
```

Comment:
I wonder if we intentionally chose a weak reference instead of `Arc` as it might avoid cyclic references (similar to how it's done for `MmArc` with `MmWeak` in coins) :thinking:

Reply:
I suspect this could be the reason of https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/12666194163/job/35297118282?pr=2301#step:9:895 failure.

Reply:
not that i can trivially find the `AtomicSwap` pointing back to it self (points to mmctx tho: https://github.com/KomodoPlatform/komodo-defi-framework/blob/0a9410258c7bf58e78d26b33b7a0ffb749c7ba16/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L213)
you see, such questions are really hard to answer in such a codebase and require digging xD
it's much easier/managable to avoid the arc/weak model and use our abortable systems to make sure cyclic refs aren't problematic.

i went for arc here tho since we already manually remove the swap after it finishes or after stopping it. since we drop the thing manually we don't need to complicate it with weak refs.

Reply:
It's worth to test that change (by using `test_mm2_stops_immediately` test) to make sure that's not the case.